impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java

```diff
@@ -129,14 +130,32 @@ public void stop() {
       }
     }
     try {
-      if(!scheduledExecutor.awaitTermination(30, SECONDS)) {
-        LOGGER.warn("Timeout while waiting on scheduler to finish");
+      while(!scheduledExecutor.awaitTermination(30, SECONDS)) {
```

Comment:
Good question. First, I was waiting for 30 seconds and that's it. Then I started thinking that getting out of the wait with task still running means we get out of `stop()` in some unknown weird state that is probably hiding a bug. So I ended up preferring to freeze the stop and log. From there, a thread dump will easily allow to debug the issue.